packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,788 @@
+          })
+          ProviderController.setChainId(chainId)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (switchError: any) {
+          if (switchError.code === 4902 || switchError.code === 5000) {
+            await walletConnectProvider.request({
+              method: 'wallet_addEthereumChain',
+              params: [
+                {
+                  chainId: `0x${chainId.toString(16)}`,
+                  rpcUrls: [NetworkRPCUrls[chainId]],
+                  chainName: NetworkNames[chainId],
+                  nativeCurrency: {
+                    name: networkCurrenySymbols[chainId],
+                    decimals: 18,
+                    symbol: networkCurrenySymbols[chainId]
+                  },
+                  blockExplorerUrls: [NetworkBlockExplorerUrls[chainId]],
+                  iconUrls: [NetworkImageIds[chainId]]
+                }
+              ]
+            })
+          } else {
+            throw new Error('Chain is not supported')
+          }
+        }
+      }
+    } else if (providerType === INJECTED_CONNECTOR_ID) {
+      const injectedProvider = provider
+      if (injectedProvider) {
+        try {
+          await injectedProvider.send('wallet_switchEthereumChain', [
+            { chainId: `0x${chainId.toString(16)}` }
+          ])
+          ProviderController.setChainId(chainId)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (switchError: any) {
+          if (switchError.code === 4902 || switchError.code === 5000) {
+            await injectedProvider.send('wallet_addEthereumChain', [
+              {
+                chainId: `0x${chainId.toString(16)}`,
+                rpcUrls: [NetworkRPCUrls[chainId]],
+                chainName: NetworkNames[chainId],
+                nativeCurrency: {
+                  name: networkCurrenySymbols[chainId],
+                  decimals: 18,
+                  symbol: networkCurrenySymbols[chainId]
+                },
+                blockExplorerUrls: [NetworkBlockExplorerUrls[chainId]],
+                iconUrls: [NetworkImageIds[chainId]]
+              }
+            ])
+          } else {
+            throw new Error('Chain is not supported')
+          }
+        }
+      }
+    } else if (providerType === EIP6963_CONNECTOR_ID) {
+      const EIP6963Provider = provider
+
+      if (EIP6963Provider) {
+        try {
+          await EIP6963Provider.send('wallet_switchEthereumChain', [
+            { chainId: `0x${chainId.toString(16)}` }
+          ])
+          ProviderController.setChainId(chainId)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (switchError: any) {
+          if (switchError.code === 4902 || switchError.code === 5000) {
+            await EIP6963Provider.send('wallet_addEthereumChain', [
+              {
+                chainId: `0x${chainId.toString(16)}`,
```

Comment:
All these payloads are repeated in each provider, can abstract them to utils as well.